### PR TITLE
NDRS-693: Fix incorrect next_finalized on Block Proposer initialization

### PR DESCRIPTION
**Pull request by fizyk20**
https://casperlabs.atlassian.net/browse/NDRS-693

### CODE REVIEW

**Review by goral09 — APPROVED** [review]
LGTM but please, test it locally first.

**Review by EdHastingsCasperLabs — APPROVED** [review]

File: node/src/components/block_proposer.rs [hunk 1]
```diff
@@ -140,10 +145,16 @@ where
...
+                    chainspec,
+                    sets,
+                    next_finalized_block,
+                },
             ) => {
                 let mut new_ready_state = BlockProposerReady {
-                    sets: sets.unwrap_or_default(),
+                    sets: sets
+                        .unwrap_or_default()
+                        .with_next_finalized(next_finalized_block),
```
Comment (afck) [inline node/src/components/block_proposer.rs:157]: So we persist that but then always override it on startup? :confused:

**Review by afck — COMMENTED** [review]
Looks good! :+1:

File: node/src/components/block_proposer.rs (hunk 1 above)
Comment (fizyk20) [reply node/src/components/block_proposer.rs:157]: Yeah. A bit suboptimal, we probably shouldn't persist it - would make more sense to take it out of the state that gets persisted.

**Pull request merged**